R/assemble.R

```diff
@@ -72,6 +72,9 @@ parse_survey <- function(surv_obj, oauth_token = getOption('sm_oauth_token'), ..
     x$choice_text[x$question_type == "multiple_choice" | x$question_subtype == "multi" & is.na(x$other_text)],
     sep = " - ")
 
+  # Fix Multiple text box question type has NA appended to the column name #66
+  x$combined_q_heading <- trimws(gsub('^\\ - NA|\\ - NA$', '', x$combined_q_heading))
+
```

Comment:
I see how this works.  But it seems like it's a band-aid after an improper paste has been made.  The block above, lines 70-73, looks long but it's basically "if the question type is multiple choice OR question type is multi and there's no "other" text, then the combined heading should get the choice text appended."  If this behavior is associated with a single question type, could we do something similar and specify that it should have a different paste combination (that doesn't include the bit that is NA)?

Maybe the best structural fix is rewriting 64-73 with this additional functionality, so that it's more explicit that if the question is of types A & B, then the combined heading is paste X-Y-Z.  If it's of type C, paste X-Y-Z-W.  Etc.  Maybe done with `case_when` and/or some tidyverse alternative to `apply`.  The fact that I use `apply` is probably evidence that this was pushing my `purrr` skills too far.

Reply:
Curious to hear what you make of that.  If that is unclear, you don't agree, or feels too hard, then we could merge this and open a new issue to make that bigger structural fix.  Or if you are up for the bigger rewrite, then go for that - maybe as a separate PR and then the simple fix below can get merged.

Reply:
I'd be happy to take a shot at a rewrite with the caveat that I might get stumped. If you're comfortable merging this fix for now, I will plan to have a better fix in the next 2 weeks or so. I'll create a new branch and submit the rewrite as a new PR.